flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/SuccessfulCheckpoint.java

```diff
@@ -64,6 +73,22 @@ public long getTimestamp() {
 		return states;
 	}
 
+	/**
+	 * Returns the task state included in the checkpoint for a given JobVertexID if it exists or 
+	 * null if no state is included for that id.
+	 * 
+	 * @param jobVertexID
+	 * @return
+	 */
+	public StateForTask getState(JobVertexID jobVertexID)
+	{
+		if(vertexToState.containsKey(jobVertexID)) {
+			return vertexToState.get(jobVertexID);
```

Comment:
I think you can save one map lookup by calling get() immediately. It will return null on a missing key.